### PR TITLE
Clean `unwrap()`s from `query-planner`

### PR DESCRIPTION
**Pull request by Enrico2**
Started from the query-planner crate.

Only remaining `unwrap()`s are in unit tests.

(also cleaned a clippy thing)

### CODE REVIEW

File: stargate/crates/query-planner/src/builder.rs [hunk 1]
```diff
@@ -364,11 +364,11 @@ fn complete_field<'a, 'q: 'a>(
         // the type_name could be a primitive type which is not in our names_to_types map.
         let return_type = context.names_to_types.get(type_name);
 
-        if return_type.is_none() || !return_type.unwrap().is_composite_type() {
+        if return_type.is_none() || !return_type.expect("checked not None").is_composite_type() {
```
Comment (ashleygwilliams) [inline stargate/crates/query-planner/src/builder.rs:367]: i think this would be better as an `if let`

File: stargate/crates/query-planner/src/context.rs [hunk 2]
```diff
@@ -214,35 +217,35 @@ impl<'q> QueryPlanningContext<'q> {
         }
 
         let return_type = self.names_to_types.get(field_def.field_type.as_name());
-        let field_type_is_not_composite =
-            return_type.is_none() || !return_type.unwrap().is_composite_type();
 
-        if field_type_is_not_composite {
-            return vec![];
-        }
+        if let Some(return_type) = return_type {
```
Comment (ashleygwilliams) [inline stargate/crates/query-planner/src/context.rs:221]: the logic here is hard to read and would benefit from being broken into some smaller functions

File: stargate/crates/query-planner/src/context.rs [hunk 3]
```diff
@@ -214,35 +217,35 @@ impl<'q> QueryPlanningContext<'q> {
...
-                    &mut res,
-                    &mut visited_fragment_names,
-                );
-                res
-            };
-            provided_fields.chain(fields).collect()
+            if let Some(provides) = self.federation.provides(field_def) {
+                let fields = {
+                    let mut visited_fragment_names: HashSet<&str> = HashSet::new();
+                    let mut res = vec![];
```
Comment (ashleygwilliams) [inline stargate/crates/query-planner/src/context.rs:234]: this smells like it would be better as a struct w/impl

File: stargate/crates/query-planner/src/context.rs [hunk 4]
```diff
@@ -214,35 +217,35 @@ impl<'q> QueryPlanningContext<'q> {
...
+                    );
+                    res
+                };
+                provided_fields.chain(fields).collect()
+            } else {
+                provided_fields.collect()
+            }
         } else {
-            provided_fields.collect()
+            vec![]
```
Comment (ashleygwilliams) [inline stargate/crates/query-planner/src/context.rs:248]: is this potentially better as a default value?

File: stargate/crates/stargate-lib/src/request_pipeline/executor.rs [hunk 5]
```diff
@@ -47,7 +47,7 @@ fn execute_node<'schema, 'request>(
     context: &'request ExecutionContext<'schema, 'request>,
     node: &'request PlanNode,
     results: &'request RwLock<Value>,
-    path: &'request ResponsePath,
+    path: &'request [String],
```
Comment (ashleygwilliams) [inline stargate/crates/stargate-lib/src/request_pipeline/executor.rs:50]: not clear why this type change? seems unrelated to removing unwraps

File: stargate/crates/stargate-lib/src/request_pipeline/executor.rs [hunk 6]
```diff
@@ -283,7 +283,7 @@ fn flatten_results_at_path<'request>(
     }
 }
 
-fn execute_selection_set(source: &Value, selections: &SelectionSet) -> Value {
+fn execute_selection_set(source: &Value, selections: &[Selection]) -> Value {
```
Comment (ashleygwilliams) [inline stargate/crates/stargate-lib/src/request_pipeline/executor.rs:286]: same here, not sure why type change is in this pr

**Review by ashleygwilliams — CHANGES_REQUESTED** [review]
there are a lot of expects here that suggest that a vec has only one item- why are we using a vec to model this data

replacing unwraps with expects is great but i would still encourage us to use proper error handling. even if we reason it is impossible, code changes, and those assumptions change. using errors and ? is much better than panicking, even if the panic has a messge.

File: stargate/crates/query-planner/src/builder.rs (hunk 1 above)
Comment (Enrico2) [reply stargate/crates/query-planner/src/builder.rs:367]: How? afaik I can't add a condition with an `||` to an `if let`

File: stargate/crates/stargate-lib/src/request_pipeline/executor.rs (hunk 5 above)
Comment (Enrico2) [reply stargate/crates/stargate-lib/src/request_pipeline/executor.rs:50]: Recommendation (that I agree with) from Clippy.
`ResponsePath` is a type alias for `Vec<String>`, using a slice `&[String]` is better than expecting a `&Vec<String>` -- it saves an allocation and allows the function to be called with an array.

**Comment by Enrico2** [comment]
I don't have the bandwidth right now to improve this code in general and refactor things out. The only purpose was to replace `unwrap`s with `expect` (and fixing that Clippy warning).

I appreciate the other suggestions but they should not block merging this.

File: stargate/crates/query-planner/src/context.rs (hunk 4 above)
Comment (abernix) [reply stargate/crates/query-planner/src/context.rs]: Tracking this in https://github.com/apollographql/federation/issues/477

File: stargate/crates/query-planner/src/context.rs (hunk 3 above)
Comment (abernix) [reply stargate/crates/query-planner/src/context.rs]: Tracking this in https://github.com/apollographql/federation/issues/477

File: stargate/crates/query-planner/src/context.rs (hunk 2 above)
Comment (abernix) [reply stargate/crates/query-planner/src/context.rs]: Tracking this in https://github.com/apollographql/federation/issues/477

**Comment by abernix** [comment]
At a high level, tracking the need to consider revisiting the `vec` in https://github.com/apollographql/federation/issues/476
